src/hfold/Parser.hs

```diff
@@ -0,0 +1,55 @@
+module Hfold.Parser where
+
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import qualified Data.Text as T
+import Text.ParserCombinators.Parsec
+
+import Hfold.Types
+
+parseFile :: Text -> [Content]
+parseFile t = case parse parseContent "content" (T.unpack t) of
+    Left err -> []
+    Right xs -> xs
+
+parseContent :: Parser [Content]
+parseContent = do
+    c <- many (quoted <|> codeBlock <|> normal)
+    eof
+    return c
+
+normal :: Parser Content
+normal = Normal <$> singleLine
+
+quoted :: Parser Content
+quoted = do
+    q <- quoteChar
+    l <- singleLine
+    return $ Quoted (q <> l)
+
+codeBlock :: Parser Content
+codeBlock = do
+    s <- codeBlockChar
+    c <- codeBlockContents
+    e <- codeBlockChar
+    eol
+    return $ CodeBlock $ s <> c <> e
+
+singleLine :: Parser Text
+singleLine = T.pack <$> manyTill anyChar (try eol)
```

Comment:
because it didn't compile without it?

I believe this function needs some kind of lookahead function, which leaves us with `try` and `lookahead`. Using `try` means the `eol` is actually consumed, as opposed to in `codeBlockContents` where we use `lookahead` because we _don't_ want to consume the triple back ticks.